### PR TITLE
[Doris On ES][Optimization] Push limit to Elasticsearch external table

### PR DESCRIPTION
**Pull request by wuyunfeng**
Apply Elasticserch terminated_after for the request which send to the Elasticsearch to achieve terminated early when limit take effect

https://github.com/apache/incubator-doris/issues/2402

### CODE REVIEW

File: be/src/exec/es/es_scan_reader.cpp [hunk 1]
```diff
@@ -49,20 +51,37 @@ ESScanReader::ESScanReader(const std::string& target, const std::map<std::string
...
     }
+
     std::string batch_size_str = props.at(KEY_BATCH_SIZE);
     _batch_size = atoi(batch_size_str.c_str());
-    _init_scroll_url = _target + REQUEST_SEPARATOR + _index + REQUEST_SEPARATOR + _type + "/_search?scroll=" + _scroll_keep_alive + REQUEST_PREFERENCE_PREFIX + _shards + "&" + REUQEST_SCROLL_FILTER_PATH;
-    _next_scroll_url = _target + REQUEST_SEARCH_SCROLL_PATH + "?" + REUQEST_SCROLL_FILTER_PATH;
+
+    if (props.find(KEY_TERMINATE_AFTER) != props.end()) {
+        _exactly_once = true;
+        _terminate_after = props.at(KEY_TERMINATE_AFTER);
```
Comment (imay) [inline be/src/exec/es/es_scan_reader.cpp]: seems that a local variable is enough for _terminate_after

File: be/src/exec/es/es_scan_reader.h [hunk 2]
```diff
@@ -73,6 +76,8 @@ class ESScanReader {
     std::string _scroll_keep_alive;
     // timeout for es http connetion
     int _http_timeout_ms;
+
+    bool _exactly_once;
```
Comment (imay) [inline be/src/exec/es/es_scan_reader.h]: 1. give it a initial value
2. add more comments to explain in what scenario is this true

File: be/src/exec/es/es_scan_reader.h [hunk 3]
```diff
@@ -65,6 +66,8 @@ class ESScanReader {
     bool _is_first;
     std::string _init_scroll_url;
     std::string _next_scroll_url;
+    std::string _search_url;
```
Comment (imay) [inline be/src/exec/es/es_scan_reader.h]: add comments to explian the difference of these url

Comment (wuyunfeng) [reply be/src/exec/es/es_scan_reader.h]: OK

File: be/src/exec/es/es_scan_reader.h (hunk 2 above)
Comment (wuyunfeng) [reply be/src/exec/es/es_scan_reader.h]: 1.  _exactly_once will be inited on the ESScanReader ctor
2. OK

File: be/src/exec/es/es_scan_reader.h (hunk 3 above)
Comment (wuyunfeng) [reply be/src/exec/es/es_scan_reader.h]: _search_url explained at the ESScanReader ctor

File: be/src/exec/es/es_scan_reader.cpp (hunk 1 above)
Comment (wuyunfeng) [reply be/src/exec/es/es_scan_reader.cpp]: Yeah.

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**